src/modules/alerting.ts

```diff
@@ -68,33 +88,29 @@ class AlertingApi extends BaseApi {
      * ```ts
      * var options = {
      *    id: 'mdExample',
+     *    type: 'info',
+     *    timeout: 3,
+     *    showCloseButton: false,
      *    markdownMessage: true
      * };
      * myClientApp.alerting.showToastPopup("Hello world", "Hello :earth_americas: How are *you* doing today?", options);
      * ```
-     * 
+     *
      * @param title - Toast title.
      * @param message - Toast Message.  Supports emoticons, emoji (unicode, shortcodes) and markdown (with markdwownMessage boolean).
      * @param options - Additonal toast options.
      *
      * @since 1.0.0
      */
-    showToastPopup(title: string, message: string, options: {
-        /**
-         * The id of the message.  Duplicate IDs will replace each other in the toast display.  All IDs will be
-         * namespaced with your app ID to avoid collisions.  (Default will just be your app's namespace and will not support multiple messages)
-         * */
-        id?: string;
-        /** Toast type, valid options are 'error', 'info', 'success'.  (Default is 'info') */
-        type?: 'error' | 'info' | 'success';
-        /** Boolean indicating if the message is in MD.  (Default is true) */
-        markdownMessage?: boolean;
-        /** Time in seconds to show the toast.  Set to 0 to disable automatic dismissal. (Default is 5) */
-        timeout?: number;
-        /** Boolean indicating if the close button should be shown. (Default is false) */
-        showCloseButton?: boolean;
-    } = {}) {
-        const messageParams: BaseToastMessageProps & typeof options = {
+    showToastPopup<
+        MessageType extends 'error' | 'info' | 'success',
+        Timeout extends number
+    >(
+        title: string,
+        message: string,
+        options: ToastOptions<MessageType, Timeout> = {} as ToastOptions<MessageType, Timeout>
```

Comment:
Should we go ahead and specify a default as `{type: 'info', timeout: 5, showCloseButton: false}`?

Reply:
Digging into this more, the defaults are actually showCloseButton: true, timeout: 7.  So, changing this would actually be a "breaking change for 2.0".  Wondering if we should just update default docs to match behavior and move on.

Reply:
So what I've got here `{}` for the default matches what we had before correct?

Reply:
Yeah, it's the same as before.  The problem is that the documented defaults are (and may have always been) incorrect.

Maybe we should handle this in another PR?  that being we should enforce/set our specified defaults here so they're not affected by downstream changes?